cobc/error.c

```diff
@@ -63,10 +70,80 @@ print_error_prefix (const char *file, int line, const char *prefix)
 	}
 }
 
+/* Display a context around the location of the error/warning, only if
+ * cb_diagnostics_show_caret is true
+
+   Only display two lines before and after. No caret yet for the column as
+   we only have the line. Since we directly use the file, source is printed
+   before any REPLACE.
+ */
+
+#define CARET_MAX_COLS 73
+static void
+diagnostics_show_caret (const char *file, int line)
+{
+	FILE* fd = fopen(file, "r");
+	if (fd == NULL) return;
+	char buffer[ CARET_MAX_COLS+1 ];
+	int line_pos = 1;
+	int char_pos = 0;
+	int printed = 0; /* nothing printed */
+	while(1){
+		int c = fgetc (fd);
+		if ( c == EOF ){
+			if (printed){
+				fprintf(stderr, "\n");
+			}
+			fclose(fd);
+			return ;
+		}
+		buffer[char_pos] = c ;
+		if (c == '\n' || char_pos == CARET_MAX_COLS){
+			buffer[char_pos] = 0;
+			if (line_pos > line-3 && line_pos < line+3){
+				if (line_pos == line-2) fprintf(stderr, "\n");
+				printed = 1;
+				fprintf (stderr, "  ");
+				if (cb_diagnostics_show_line_numbers){
+					fprintf (stderr, "%04d ",
```

Comment:
Currently that does not only skip line number but the whole printing, no?

Reply:
Wrong indentation, sorry...